### PR TITLE
Mikhail/ch6837/create skeleton application

### PR DESCRIPTION
**Pull request by mikhailsakhnov**
https://app.clubhouse.io/callstatsio/story/6837/create-skeleton-application

At this point theere is no business related logic in this application, only infrastructure:
- vault integration
- database connection 
- health_check
- grpc service exists but doesn't implemented
- http status handler

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	}
+	mc, err := mongo.NewStandardClient(app.VaultClient(), mongoOpts)
+	if err != nil {
+		logger.Panic("Failed to create mongo StandardClient", log.Error(err))
+	}
+
+	mongoStorage := storage.NewMongoStorage(mc)
+
+	app.WithHTTPPort(settings.HTTPStatusPort).
+		ServeHTTP(http.SetupStatusRouter(mongoStorage.Check))
```
[the remark this answers is not in the document]
Comment (mikhailsakhnov) [reply main.go]: Nope, it shouldn't (it has go statement inside)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ADD vendor/ vendor/
+ADD scripts/docker/install_service.sh install_service.sh
+ADD main.go main.go
+ADD src/ src/
+ADD creds creds
+RUN bash ./install_service.sh $SERVICE_VERSION github.com/callstats-io/conference-data-service/src/config
+
+EXPOSE 14000 14001
+
+CMD /go/bin/conference-data-service
```
[the remark this answers is not in the document]
Comment (mikhailsakhnov) [reply Dockerfile]: Ok, fixed editor settings

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
+version: '2'
```
[the remark this answers is not in the document]
Comment (mikhailsakhnov) [reply docker-compose.yml]: Is it critical? To remove version I need to rewrite build section in every target

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      - VAULT_SKIP_VERIFY=true
+      - VAULT_ENABLE_MONGO=true
+      - VAULT_MONGO_CREDS_PATH=dev/mongodb/devc1/creds/conference_data_service
+      - VAULT_TEST_BOOTSTRAP_TOKEN=7d52e719-c3bb-4fd0-9f08-6e11051fea19
+    command: '/go/bin/conference-data-service'
+
+  test_conference_data_service:
+    build:
+      context: .
+      dockerfile: Dockerfile.test
```
[the remark this answers is not in the document]
Comment (mikhailsakhnov) [reply docker-compose.yml]: FIxed (thanks to default .gitignore)

File: src/http/status_handler.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+// Status responds if the system is functional or not, TODO: healthcheck should be async
+func (sh *StatusHandler) Status(w http.ResponseWriter, r *http.Request) {
+	status := &Status{
+		Status:    StatusUp,
+		BuildTime: config.ImageBuildTime,
+		Version:   config.ServiceVersion,
+	}
+
+	if err := sh.mongoChecker(r.Context()); err != nil {
+		fmt.Println(err)
```
Comment (stenhagglund) [inline src/http/status_handler.go]: Change to log or remove this.

File: src/http/status_handler.go [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+	}
+}
+
+// SetupStatusRouter sets up a router for status handling and metrics
+func SetupStatusRouter(mongoChecker func(ctx context.Context) error) *httptreemux.TreeMux {
+	router := httptreemux.New()
+	sh := NewStatusHandler(mongoChecker)
+	baseMiddlewares := middleware.NewConfiguratorWithCapacity(10)
+	baseMiddlewares.Use(middleware.RequestContextLogger())
+	baseMiddlewares.Use(middleware.RequestLifetimeLogger())
```
Comment (stenhagglund) [inline src/http/status_handler.go]: You don't likely need these middleware for status endpoint and other internal ones. RequestContextLogger adds a requestID to the requests which is useful to correlate messages (not sure if that's useful for status) and lifetimelogger is not needed with prometheus (just logs start + finish message for every request).

**Pull request merged**